src/app/services/case/case.service.ts

```diff
@@ -0,0 +1,78 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { ErrorHandlerService } from '../error/error-handler.service';
+import { CaseData } from '../../../app/types/case';
+
+const GET_CASE_PATH = '/api/cases/';
+const ADVANCED_SEARCH_CASE_PATH = '/api/cases/search';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CaseService {
+  constructor(private readonly http: HttpClient, private ErrorHandler: ErrorHandlerService) {}
+
+  //Single get case API
+  getCase(caseId: string | number): Observable<CaseData> {
+    const apiURL = `${GET_CASE_PATH}?caseId=${caseId}`;
+    return this.http.get<CaseData>(apiURL).pipe(
+      catchError((err: Error) => {
+        this.ErrorHandler.handleError(err);
+        return throwError(() => err);
+      })
+    );
+  }
+
+  //Advanced search API fetching multiple cases
+  getCasesAdvanced(
+    case_number?: string | null,
+    courthouse?: string | null,
+    courtroom?: string | null,
+    judge_name?: string | null,
+    defendant_name?: string | null,
+    date_from?: string | null,
+    date_to?: string | null,
+    event_text_contains?: string | null
+  ): Observable<CaseData[]> {
+    let params = new HttpParams();
+
+    //Process optional parameters to form HttpParams
+    if (case_number) {
+      params = params.set('case_number', case_number);
+    }
+    if (courthouse) {
+      params = params.set('courthouse', courthouse);
+    }
+    if (courtroom) {
+      params = params.set('courtroom', courtroom);
+    }
+    if (judge_name) {
+      params = params.set('judge_name', judge_name);
+    }
+    if (defendant_name) {
+      params = params.set('defendant_name', defendant_name);
+    }
+    if (date_from) {
+      params = params.set('date_from', date_from);
+    }
+    if (date_to) {
+      params = params.set('date_to', date_to);
+    }
+    if (event_text_contains) {
+      params = params.set('event_text_contains', event_text_contains);
+    }
+
+    const options = { params: params };
+    const apiURL = `${ADVANCED_SEARCH_CASE_PATH}`;
+
+    //Make API call out to advanced search case API, return Observable to Case Component, catch any errors and pass to ErrorHandler service
+    return this.http.get<CaseData[]>(apiURL, options).pipe(
+      catchError((err: Error) => {
+        this.ErrorHandler.handleError(err);
+        return throwError(() => err);
```

Comment:
What displays on the page if this request errors?

Reply:
I will look to present a generic error message if a non expected error is detected, but it might be worth exploring generic handling of errors in an interceptor or similar to reroute to an error page? Not sure what the standard is for HMCTS on this